### PR TITLE
Address JavaDoc issues

### PR DESCRIPTION
**Pull request by thc202**
Address JavaDoc issues in some classes, tidy up class JavaDoc, add
missing docs on methods and parameters, replace closing HTML tags with
starting tags and remove empty docs of overridden methods.

### CODE REVIEW

File: src/org/parosproxy/paros/extension/ExtensionHookMenu.java [hunk 1]
```diff
@@ -98,8 +99,9 @@
 
     
     /**
-     * Get the plugin popup menu used for the whole workbench.
-     * @return
+     * Gets the popup menu items used for the whole workbench.
+     * 
+     * @return a {@code List} containing the pop up menu items of the extension
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/ExtensionHookMenu.java]: Popup should be all one word here for consistency

File: src/org/parosproxy/paros/extension/ExtensionHookMenu.java [hunk 2]
```diff
@@ -195,7 +202,7 @@ public void addNewMenu(JMenu menu) {
     /**
      * Add a popup menu item used for the whole workbench.  Conditions can be set in PluginMenu
      * when the popup menu can be used.
-     * @param menuItem
+     * @param menuItem the pop up menu item
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/ExtensionHookMenu.java]: Same here

File: src/org/parosproxy/paros/extension/option/DatabaseParam.java [hunk 3]
```diff
@@ -215,16 +215,24 @@ public void setResponseBodySize(int responsebodysize) {
     }
 
     /**
-     * Gets the new session option
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/option/DatabaseParam.java]: Should this file have a ZAP comment

File: src/org/parosproxy/paros/network/HttpRequestHeader.java [hunk 4]
```diff
@@ -509,8 +508,8 @@ public boolean isImage() {
      * Return if the data given is a request header basing on the first start
      * line.
      *
-     * @param data
-     * @return
+     * @param data the data to be checked
+     * @return {@code true} if the data contains a
```
Comment (kingthorin) [inline src/org/parosproxy/paros/network/HttpRequestHeader.java]: Contains a ???

File: src/org/zaproxy/zap/model/Context.java [hunk 5]
```diff
@@ -296,10 +296,10 @@ private void fillNodesInContext(SiteNode rootNode, List<SiteNode> nodesList) {
...
 
 	/**
-	 * Fills a given list with nodes in scope, searching recursively.
+	 * Tells whether or not the given node or any of its child nodes are in context.
 	 * 
-	 * @param rootNode the root node
-	 * @param nodesList the nodes list
+	 * @param node the node to start the check
+	 * @return {@code true} if at lest one node is in context, {@code false} otherwise
 	 */
```
Comment (kingthorin) [inline src/org/zaproxy/zap/model/Context.java]: Least not lest :)

File: src/org/parosproxy/paros/extension/option/DatabaseParam.java (hunk 3 above)
Comment (thc202) [inline src/org/parosproxy/paros/extension/option/DatabaseParam.java]: It shouldn't, it's not a Paros class.

**Comment by thc202** [comment]
Comments addressed, thanks!

File: src/org/parosproxy/paros/extension/option/DatabaseParam.java (hunk 3 above)
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/option/DatabaseParam.java]: Ok. I was trusting the path, but I guess 'extension' should have tipped me off

**Comment by kingthorin** [comment]
Looks good

**Pull request merged**